### ISSUE TITLE
CFFI broken on `3.12.0rc3`

### ISSUE THREAD

**Issue by dhirschfeld** [conda-forge/python-feedstock#649]
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

With `3.12.0rc3` from `conda-forge` I get an access violation when trying to load a DLL on `win-64`:
```
(quantdev)
❯ python
Python 3.10.11 | packaged by conda-forge | (main, May 10 2023, 18:51:25) [MSC v.1934 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> from cffi import FFI
>>> ffi = FFI()
>>> ffi.dlopen('ucrtbase')
<cffi.api._make_ffi_library.<locals>.FFILibrary object at 0x00000211AF4AC970>
>>> exit()

❯ mamba activate test312
(test312)
❯ python -X faulthandler
Python 3.12.0rc3 | packaged by conda-forge | (main, Sep 19 2023, 13:41:36) [MSC v.1935 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> from cffi import FFI
>>> ffi = FFI()
>>> ffi.dlopen('ucrtbase')
Windows fatal exception: access violation

Current thread 0x00000e10 (most recent call first):
Windows fatal exception: access violation

NativeCommandExitException: Program "python.exe" ended with non-zero exit code: -1073740791.
```

***xref:*** https://foss.heptapod.net/pypy/cffi/-/issues/574

### Installed packages

```shell
❯ mamba list
List of packages in environment: "C:\\conda\\envs\\test312"

  Name              Version       Build                   Channel
---------------------------------------------------------------------------------------
  attrs             23.1.0        pyh71513ae_1            conda-forge
  bzip2             1.0.8         h8ffe710_4              conda-forge
  ca-certificates   2023.7.22     h56e8100_0              conda-forge
  cffi              1.15.1        py312he70551f_5         conda-forge
  idna              3.4           pyhd8ed1ab_0            conda-forge
  libexpat          2.5.0         h63175ca_1              conda-forge
  libffi            3.4.2         h8ffe710_5              conda-forge
  libsqlite         3.43.0        hcfcfb64_0              conda-forge
  libzlib           1.2.13        hcfcfb64_5              conda-forge
  openssl           3.1.3         hcfcfb64_0              conda-forge
  outcome           1.2.0         pyhd8ed1ab_0            conda-forge
  pip               23.2.1        pyhd8ed1ab_0            conda-forge
  pycparser         2.21          pyhd8ed1ab_0            conda-forge
  python            3.12.0rc3     rc3_h8cdc648_0_cpython  conda-forge/label/python_rc
  python_abi        3.12          4_cp312                 conda-forge
  setuptools        68.2.2        pyhd8ed1ab_0            conda-forge
  sniffio           1.3.0         pyhd8ed1ab_0            conda-forge
  sortedcontainers  2.4.0         pyhd8ed1ab_0            conda-forge
  tk                8.6.13        hcfcfb64_0              conda-forge
  trio              0.22.2        py312h2e8e312_1         local
  tzdata            2023c         h71feb2d_0              conda-forge
  ucrt              10.0.22621.0  h57928b3_0              conda-forge
  vc                14.3          h64f974e_17             conda-forge
  vc14_runtime      14.36.32532   hdcecf7f_17             conda-forge
  vs2015_runtime    14.36.32532   h05e6639_17             conda-forge
  wheel             0.41.2        pyhd8ed1ab_0            conda-forge
  xz                5.2.6         h8d14728_0              conda-forge
```

### Environment info

```shell
❯ mamba info

       libmamba version : 1.4.9
     micromamba version : 1.4.9
           curl version : libcurl/8.1.2-DEV Schannel zlib/1.2.13
     libarchive version : libarchive 3.6.2 zlib/1.2.13 liblzma/5.4.3 bz2lib/1.0.8 liblz4/1.9.4 libzstd/1.5.5
       envs directories : C:\conda\envs
          package cache : C:\conda\pkgs
                          C:\conda\pkgs
            environment : test312 (active)
           env location : C:\conda\envs\test312
      user config files : C:\Users\dhirschfeld\.mambarc
 populated config files : C:\Users\dhirschfeld\.condarc
                          C:\conda\.mambarc
       virtual packages : __win=0=0
                          __archspec=1=x86_64
                          __cuda=12.2=0
               channels : https://conda.anaconda.org/conda-forge/win-64
                          https://conda.anaconda.org/conda-forge/noarch
       base environment : C:\conda
               platform : win-64
```

**Comment by dhirschfeld** [comment]
If I manually uninstall `cffi==1.15.1` and instead install `cffi==1.16.0rc2` it appears to work:
```
(test312)
❯ python
Python 3.12.0rc3 | packaged by conda-forge | (main, Sep 19 2023, 13:41:36) [MSC v.1935 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import cffi
>>> cffi.__version__
'1.16.0rc2'
>>> from cffi import FFI
>>> ffi = FFI()
>>> ffi.dlopen('ucrtbase')
<cffi.api._make_ffi_library.<locals>.FFILibrary object at 0x000002409681A420>
>>>
```

**Comment by h-vetinari** [comment]
Cool, thanks. Could you open a PR to the cffi-feedstock that adds a test based on
```
>>> from cffi import FFI
>>> ffi = FFI()
>>> ffi.dlopen('ucrtbase')
```
on windows? On unix, we'd have to use some other library, perhaps glibc?

**Comment by dhirschfeld** [comment]
Will do. Will have to wait until tonight though - have to switch to `$dayjob` now :D

**Comment by jakirkham** [comment]
Did PR ( https://github.com/conda-forge/cffi-feedstock/pull/52 ) fix this?

**Comment by dhirschfeld** [comment]
It appears this has been fixed in the just released python 3.12. Can always re-open if there's anything further that needs to be addressed.
